interop-scoring/main.js

```diff
@@ -0,0 +1,506 @@
+    }
+  }
+  if (hadErrors) {
+    throw new Error('Missing data for some runs (see errors logged above). ' +
+        'Try running "git fetch --all --tags" in results-analysis-cache/');
+  }
+
+  // Load the test result trees into memory; creates a list of recursive tree
+  // structures: tree = { trees: [...], tests: [...] }. Each 'tree' represents a
+  // directory, each 'test' is the results from a given test file.
+  console.log('Iterating over all runs, loading test results');
+  before = Date.now();
+  for (const runs of alignedRuns.values()) {
+    for (const run of runs) {
+      // Just in case someone ever adds a 'tree' field to the JSON.
+      if (run.tree) {
+        throw new Error('Run JSON contains "tree" field; code needs changed.');
+      }
+      run.tree = await lib.results.getGitTree(repo, run);
+    }
+  }
+  after = Date.now();
+  console.log(`Loading ${alignedRuns.size} sets of runs took ` +
+      `${after - before} ms`);
+
+  const dateToScoresMaps = new Map();
+
+  // Map from labels to tests (includes)
+  const labeledTests = new Map();
+  const url = 'https://wpt.fyi/api/metadata?includeTestLevel=true&product=chrome';
+  const response = await fetch(url);
+  const metadata = await response.json();
+  for (const [test, metadataList] of Object.entries(metadata)) {
+    for (const {label} of metadataList) {
+      if (label) {
+        if (!labeledTests.has(label)) {
+          labeledTests.set(label, new Set());
+        }
+        labeledTests.get(label).add(test);
+      }
+    }
+  }
+  // category is an object with "name" and "labels" props.
+  for (const category of categories) {
+    console.log(`Scoring runs for ${category.name}`);
+    const testsSet = new Set();
+    // We aggregate all the tests with the labels defined in the category.
+    for (const label of category.labels) {
+      const labeledTestsSet = labeledTests.get(label);
+      if (!labeledTestsSet || !labeledTestsSet.size) {
+        throw new Error(`No tests labeled for ${label}`);
+      }
+      // Keep a unique set of tests associated with the category.
+      labeledTestsSet.forEach(test => testsSet.add(test));
+    }
+    const dateToScores = await scoreCategory(category, experimental, products,
+        alignedRuns, testsSet);
+    // Store the entire dateToScores for producing the unified CSV later.
+    dateToScoresMaps.set(category.name, dateToScores);
+  }
+
+  // TODO: Once the other score CSVs are no longer used, we can push
+  // some of this logic into scoreCategory and simplify things.
+  let unifiedCsv = 'date';
+  for (const product of products) {
+    const categoryLabels = categories.map(c => `${product}-${c.name}`);
+    unifiedCsv += `,${product}-version,${categoryLabels.join()}`;
+  }
+  // Add the interop category headers.
+  // An arbitrary interop version header is kept for ease of parsing on the interop dashboard.
+  unifiedCsv += `,interop-version,${categories.map(c => `interop-${c.name}`)}`;
+  unifiedCsv += '\n';
```

Comment:
The interop scores for each category are added as columns with the `interop-` prefix, similar to each browser.